src/test/coins_tests.cpp

```diff
@@ -396,60 +394,69 @@ BOOST_AUTO_TEST_CASE(anchor_pop_regression_test)
         CCoinsViewCacheTest cache1(&base);
 
         // Create dummy anchor/commitment
-        ZCIncrementalMerkleTree tree;
-        uint256 cm = GetRandHash();
-        tree.append(cm);
+        Tree tree;
+        tree.append(GetRandHash());
 
         // Add the anchor and flush to disk
-        cache1.PushSproutAnchor(tree);
+        cache1.PushAnchor(tree);
         cache1.Flush();
 
         // Remove the anchor, but don't flush yet!
-        cache1.PopAnchor(ZCIncrementalMerkleTree::empty_root(), SPROUT);
+        cache1.PopAnchor(Tree::empty_root(), type);
 
         {
             CCoinsViewCacheTest cache2(&cache1); // Build cache on top
-            cache2.PushSproutAnchor(tree); // Put the same anchor back!
+            cache2.PushAnchor(tree); // Put the same anchor back!
             cache2.Flush(); // Flush to cache1
         }
 
         // cache2's flush kinda worked, i.e. cache1 thinks the
         // tree is there, but it didn't bring down the correct
         // treestate...
         {
-            ZCIncrementalMerkleTree checktree;
-            BOOST_CHECK(cache1.GetSproutAnchorAt(tree.root(), checktree));
+            Tree checktree;
+            BOOST_CHECK(GetAnchorAt(cache1, tree.root(), checktree));
             BOOST_CHECK(checktree.root() == tree.root()); // Oh, shucks.
         }
 
         // Flushing cache won't help either, just makes the inconsistency
         // permanent.
         cache1.Flush();
         {
-            ZCIncrementalMerkleTree checktree;
-            BOOST_CHECK(cache1.GetSproutAnchorAt(tree.root(), checktree));
+            Tree checktree;
+            BOOST_CHECK(GetAnchorAt(cache1, tree.root(), checktree));
             BOOST_CHECK(checktree.root() == tree.root()); // Oh, shucks.
         }
     }
 }
 
-BOOST_AUTO_TEST_CASE(anchor_regression_test)
+BOOST_AUTO_TEST_CASE(anchor_pop_regression_test)
+{
+    BOOST_TEST_CONTEXT("Sprout") {
+        anchorPopRegressionTestImpl<ZCIncrementalMerkleTree>(SPROUT);
+    }
+    BOOST_TEST_CONTEXT("Sprout") {
```

Comment:
`Sapling` (not sure why my previous comment about this was marked as obsolete...)